components/script/dom/globalscope.rs

```diff
@@ -2227,6 +2231,66 @@ impl GlobalScope {
         }))
     }
 
+    pub fn create_image_bitmap(
+        &self,
+        image: ImageBitmapSource,
+        options: &ImageBitmapOptions,
+    ) -> Rc<Promise> {
+        let global = self.global();
+        let in_realm_proof = AlreadyInRealm::assert(&global);
+        let p = Promise::new_in_current_realm(&global, InRealm::Already(&in_realm_proof));
+        if options.resizeWidth.unwrap() == 0 {
+            p.reject_error(Error::InvalidState)
+        }
+
+        if options.resizeHeight.unwrap() == 0 {
+            p.reject_error(Error::InvalidState)
+        }
+
+        let image_bitmap = ImageBitmap::new(&global, 0, 0).unwrap();
+
+        let promise = match image {
+            ImageBitmapSource::HTMLCanvasElement(ref canvas) => {
+                // https://html.spec.whatwg.org/multipage/#check-the-usability-of-the-image-argument
+                if !canvas.is_valid() {
+                    p.reject_error(Error::InvalidState)
+                }
+
+                if let Some((data, size)) = canvas.fetch_all_data() {
+                    let data = DomRefCell::new(
+                        data.map(|data| data.to_vec())
+                            .unwrap_or_else(|| vec![0; size.area() as usize * 4]),
+                    );
+
+                    image_bitmap.set_bitmap_data(data);
+                    image_bitmap.set_origin_clean(canvas.origin_is_clean());
+                    p.resolve_native(&(image_bitmap));
+                }
+                p
+            },
+            ImageBitmapSource::OffscreenCanvas(ref canvas) => {
+                // https://html.spec.whatwg.org/multipage/#check-the-usability-of-the-image-argument
+                if !canvas.is_valid() {
+                    p.reject_error(Error::InvalidState)
+                }
+
+                if let Some((data, size)) = canvas.fetch_all_data() {
+                    let data = DomRefCell::new(
+                        data.map(|data| data.to_vec())
+                            .unwrap_or_else(|| vec![0; size.area() as usize * 4]),
+                    );
+
+                    image_bitmap.set_bitmap_data(data);
+                    image_bitmap.set_origin_clean(canvas.origin_is_clean());
+                    p.resolve_native(&(image_bitmap));
+                }
+                p
+            },
+            _ => p,
```

Comment:
Let's use `p.reject_error(Error::NotSupported)` so we don't cause timeouts in tests that will otherwise endlessly wait for something that will never happen.

Reply:
Added this and then returned p.